test/proto/deprecated.proto

```diff
@@ -2,14 +2,16 @@ syntax = "proto3";
 
 package envoy.test.deprecation_test;
 
+import "envoy/annotations/deprecation.proto";
+
 message Base {
   string not_deprecated = 1;
   string is_deprecated = 2 [deprecated = true];
-  string is_deprecated_fatal = 3 [deprecated = true];
+  string is_deprecated_fatal = 3 [deprecated = true, (envoy.annotations.do_not_use) = true];
```

Comment:
So, IIUC, your tooling or manual deprecations will set this on v2. As long as protoxform doesn't delete them (which it might today, ping me tomorrow on how to avoid this), we're good for v3, since the field doesn't exist there.